engine/execution/ingestion/engine.go

```diff
@@ -440,6 +458,54 @@ func (e *Engine) BlockProcessable(b *flow.Header) {
 	}
 }
 
+func (e *Engine) checkAndStopExecution(header *flow.Header, stopHeight uint64, stopCrash bool) {
+	executed, err := state.IsBlockExecuted(e.unit.Ctx(), e.execState, header.ParentID)
+	if err != nil {
+		e.log.Error().Err(err).Str("block_id", header.ID().String()).Msg("failed to check if the block has been executed")
+		return
+	}
```

Comment:
### ❓❗ 
To me, this looks like a case of "log error and continue with best effort", which is an anti-pattern (-> [conventions](https://github.com/onflow/flow-go/blob/master/CodingConventions.md#anti-pattern)). In case it is actually possible to continue here and the error is _not_ and indicator of internal state corruption, please add a comment explaining why this is the case.